checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -189,18 +190,19 @@ public AnnotationMirror aliasedAnnotation(AnnotationMirror a) {
      * @param qual The qualifier to investigate.
      */
     private void addUnitsRelations(Class<? extends Annotation> qual) {
+        // get the annotation mirror
         AnnotationMirror am = AnnotationUtils.fromClass(elements, qual);
-
+        // get all the meta annotations of the class
         for (AnnotationMirror ama : am.getAnnotationType().asElement().getAnnotationMirrors() ) {
-            if (ama.getAnnotationType().toString().equals(UnitsRelations.class.getCanonicalName())) {
+            // if there's a UnitsRelations meta annotation
+            if (ama.getAnnotationType().toString().equals(org.checkerframework.checker.units.qual.UnitsRelations.class.getCanonicalName())) {
                 @SuppressWarnings("unchecked")
-                Class<? extends UnitsRelations> theclass = (Class<? extends UnitsRelations>)
-                    AnnotationUtils.getElementValueClass(ama, "value", true);
+                Class<? extends UnitsRelations> theclass = (Class<? extends UnitsRelations>) AnnotationUtils.getElementValueClass(ama, "value", true);
                 String classname = theclass.getCanonicalName();
 
                 if (!getUnitsRel().containsKey(classname)) {
                     try {
-                        unitsRel.put(classname, ((UnitsRelations) theclass.newInstance()).init(processingEnv));
+                        getUnitsRel().put(classname, ((UnitsRelations) theclass.newInstance()).init(processingEnv));
```

Comment:
Why is this call of getUnitsRel necessary? Two lines above it is already called, so we're sure the field was initialized.